File: frontend/src/services/analytics.ts
import api from './api';
import type { ApiResponse } from '@/types';

export interface AnalyticsOverview {
  total_contacts: number;
  total_won: number;
  total_lost: number;
  overall_conversion_rate: number;
  total_deal_amount: number;
  avg_deal_value: number;
}

export interface ConversionPoint {
  date: string;
  total: number;
  won: number;
  rate: number;
}

export interface ChannelDistribution {
  channel: string;
  count: number;
  percentage: number;
}

export interface SalesRanking {
  user_id: string;
  user_name: string;
  deal_count: number;
  deal_amount: number;
  conversion_rate: number;
}

export interface AnalyticsDashboard {
  overview: AnalyticsOverview;
  conversion_trend: ConversionPoint[];
  channel_distribution: ChannelDistribution[];
  sales_ranking: SalesRanking[];
}

export interface SalesTargetItem {
  id: string;
  user_id: string;
  user_name: string | null;
  year: number;
  month: number;
  target_amount: number;
  target_count: number;
  created_at: string;
  updated_at: string;
}

export const analyticsApi = {
  getAnalytics: (days = 90) =>
    api.get<ApiResponse<AnalyticsDashboard>>('/analytics', { params: { days } }),
};

export const salesTargetsApi = {
  getTargets: (params?: { year?: number; user_id?: string }) =>
    api.get<ApiResponse<SalesTargetItem[]>>('/sales-targets', { params }),

  createTarget: (data: { user_id: string; year: number; month: number; target_amount: number; target_count: number }) =>
    api.post<ApiResponse<SalesTargetItem>>('/sales-targets', data),

  updateTarget: (id: string, data: { target_amount?: number; target_count?: number }) =>
    api.put<ApiResponse<SalesTargetItem>>(`/sales-targets/${id}`, data),

  deleteTarget: (id: string) =>
    api.delete<ApiResponse<null>>(`/sales-targets/${id}`),
};

File: backend/app/schemas/analytics.py
from typing import Optional
from pydantic import BaseModel


class ConversionPoint(BaseModel):
    """转化率趋势数据点"""
    date: str
    new_contacts: int
    won: int


class ChannelDistribution(BaseModel):
    """渠道分布"""
    channel: str
    count: int
    percentage: float


class SalesRanking(BaseModel):
    """销售排名"""
    user_id: str
    user_name: str
    deal_count: int
    deal_amount: float
    conversion_rate: float


class AnalyticsOverview(BaseModel):
    """分析概览"""
    total_contacts: int
    total_won: int
    total_lost: int
    overall_conversion_rate: float
    total_deal_amount: float
    avg_deal_value: float


class AnalyticsDashboard(BaseModel):
    """完整分析报表"""
    overview: AnalyticsOverview
    conversion_trend: list[ConversionPoint]
    channel_distribution: list[ChannelDistribution]
    sales_ranking: list[SalesRanking]
